griptape/drivers/prompt/text_gen_driver.py

```diff
@@ -0,0 +1,43 @@
+import copy
+import requests
+from attr import define, field
+
+from griptape.artifacts import TextArtifact
+from griptape.drivers import BasePromptDriver
+
+
+@define
+class TextGenPromptDriver(BasePromptDriver):
+    """
+    Prompt Driver for the text-generation-webui https://github.com/oobabooga/text-generation-webui
+
+    params usage example:
+    https://github.com/oobabooga/text-generation-webui/blob/main/api-examples/api-example.py
+    """
+    preset: str = field(default=None, kw_only=True)
+    params: dict = field(factory=dict, kw_only=True)
+
+    model_url: str = field(default="http://localhost:5000", kw_only=True)
+    generate_uri: str = field(default="/api/v1/generate", kw_only=True)
+
+    model = field(default=None, kw_only=True)
+    tokenizer = field(default=None, kw_only=True)
```

Comment:
The tokenizer shouldn't default to `None` here as it's required by other parts of the framework. We can make it a required param instead.

Reply:
i wasn't sure what to set here, i saw others do have, but i dont know which one to set for text gen, based on what the tokenizer is paired? for example for opanai why tiktoken is used?

Reply:
The tokenizer here should match the tokenizer the underlying model is using. OpenAI prompt driver is paired with tiktoken because that's what all OpenAI models use.